mllib/src/main/scala/org/apache/spark/mllib/clustering/BisectingKMeans.scala

```diff
@@ -156,20 +157,34 @@ class BisectingKMeans private (
   private[spark] def run(
       input: RDD[Vector],
       instr: Option[Instrumentation]): BisectingKMeansModel = {
+    val instances: RDD[(Vector, Double)] = input.map {
+      case (point) => (point, 1.0)
+    }
+    runWithWeight(instances, None)
+  }
+
+  private[spark] def runWithWeight(
+      input: RDD[(Vector, Double)],
+      instr: Option[Instrumentation]): BisectingKMeansModel = {
     if (input.getStorageLevel == StorageLevel.NONE) {
       logWarning(s"The input RDD ${input.id} is not directly cached, which may hurt performance if"
         + " its parent RDDs are also not cached.")
     }
-    val d = input.map(_.size).first()
+    val d = input.map( i => i._1.size).first()
     logInfo(s"Feature dimension: $d.")
 
+    val dataVectorWithNorm = input.map(d => d._1)
```

Comment:
It seems that var `dataVectorWithNorm` is only used to get `norms`, so what about removing it and make `val norms = input.map(d => Vectors.norm(d._1, 2.0))...`